operator/controllers/agent/deployment.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agent
+
+import (
+	"fmt"
+
+	"github.com/fluxninja/aperture/v2/operator/controllers"
+
+	"github.com/go-logr/logr"
+	"github.com/imdario/mergo"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	agentv1alpha1 "github.com/fluxninja/aperture/v2/operator/api/agent/v1alpha1"
+)
+
+// deploymentForAgent prepares the Deployment object for the Agent.
+func deploymentForAgent(instance *agentv1alpha1.Agent, log logr.Logger, scheme *runtime.Scheme) (*appsv1.Deployment, error) {
+	spec := instance.Spec
+
+	podLabels := controllers.CommonLabels(spec.Labels, instance.GetName(), controllers.AgentServiceName)
+	if spec.PodLabels != nil {
+		if err := mergo.Map(&podLabels, spec.PodLabels, mergo.WithOverride); err != nil {
+			log.Info(fmt.Sprintf("failed to merge the Pod labels for Deployment. error: %s.", err.Error()))
```

Comment:
If this is error, should we log it on error level and return?

Reply:
This won't happen as we are merging maps but as golangci-lint was complaining I added a simple log